src/logics/Logic.h

```diff
@@ -164,7 +164,8 @@ class Logic {
     virtual PTRef mkConst(char const *);
     virtual PTRef mkConst(SRef, char const *);
 
-    SymRef declareFun(std::string const & fname, SRef rsort, vec<SRef> const & args, SymbolConfig const & symbolConfig);
+    SymRef declareFun(std::string const & fname, SRef rsort, vec<SRef> const & args, SymbolConfig const & symbolConfig,
+                      bool duplicate = false);
```

Comment:
1. I don't necessarily see why it is a problem, it gives more possibilities to the outside users potentially, but also for now it wouldn't allow to do anything new with the API I think...
2. Can do this as well, but it also will be function in `Logic.h` in the end, don't have a strong opinion on this though. @aehyvari  @blishko  guys what do you think?
3. I'm not exactly getting your idea here... You're suggesting we should return a vector of symbols? But now we are not storing duplicate functions SymRefs anywhere I think... `NONLIN_MUL` is an exception because it is stored as a part of the `ArithLogic` SymRefs, but it is only created and used withing `ArithLogic` at the moment...

Reply:
After a discussion with @BritikovKI, I think we should leave the previous API unchanged and add a protected special function that would allow to create an internal `SymRef` that is derived from an already existing, publicly accessible one. So, in the case of integers, we would first create `sym_Int_TIMES` and then internally two auxiliary symbols `sym_Int_TIMES_LIN` and `sym_Int_TIMES_NONLIN` using a special function that would use `sym_Int_TIMES` as an argument. Although the implementation may not make an explicit connection between the auxiliary symbols and the symbol `sym_Int_TIMES`, I would still use it as a parameter to clarify what is going on. At the moment, it may be sufficient to represent the auxiliary symbols as `SymRef` while not including them in the "public" symbol table.